src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,6 +1596,1436 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
     ObjectGuid _playerGUID;
 };
 
+enum SaltyRocka
+{
+    NPC_SALTY_GORGAR                            = 45497,
+
+    EVENT_ROCKA_CHECK_CONVERSATION              = 1,
+    EVENT_ROCKA_CHOOSE_CONVERSATION             = 2,
+    EVENT_ROCKA_CONVERSATION_COOLDOWN           = 3,
+    EVENT_ROCKA_TALK                            = 4,
+
+    TALK_ROCKA_0                                = 0,
+    TALK_ROCKA_1                                = 1,
+    TALK_ROCKA_2                                = 2,
+    TALK_ROCKA_3                                = 3,
+    TALK_ROCKA_4                                = 4,
+    TALK_ROCKA_5                                = 5,
+    TALK_ROCKA_6                                = 6,
+    TALK_ROCKA_7                                = 7,
+    TALK_ROCKA_8                                = 8,
+    TALK_ROCKA_9                                = 9,
+    TALK_GORGAR_0                               = 0,
+    TALK_GORGAR_1                               = 1,
+    TALK_GORGAR_2                               = 2,
+    TALK_GORGAR_3                               = 3,
+    TALK_GORGAR_4                               = 4,
+    TALK_GORGAR_5                               = 5
+};
+
+// 45498 - "Salty" Rocka
+struct npc_silverpine_salty_rocka : public ScriptedAI
+{
+    npc_silverpine_salty_rocka(Creature* creature) : ScriptedAI(creature), _playerNear(false) { }
+
+    void JustAppeared() override
+    {
+        CheckForGorgor();
+    }
+
+    void Reset() override
+    {
+        _events.Reset();
+        _events.ScheduleEvent(EVENT_ROCKA_CHECK_CONVERSATION, 1s);
+    }
+
+    void MoveInLineOfSight(Unit* who) override
+    {
+        if (who->GetTypeId() == TYPEID_PLAYER)
```

Comment:
i think this won't really work if we have > 1 player. Imagine the following scenario:
- you run into range 25.0f => you become `_playerGUID` meanwhile
- i run into LoS range => i become `_playerGUID` in that case
- you run into < 20.0f range => nothing happens because i'm `_playerGUID`
- i run into < 20.0f range => convo happens

Reply:
Feedback applied. I went with a more direct approach: if any player moves in the line of sight within 20 yards, it will check if the conversation is on CD; if it's not, it will start the conversation. There's no need to check constantly in an infinite loop of check the cooldown is reset after a certain time after being triggered.